test/query/shorthand.test.ts

```diff
@@ -33,6 +35,124 @@ describe('query/shorthand', () => {
+        'point|calculate:{b2:3*datum["b2"]}|filter:"datum[\\"b2\\"] > 60"|filterInvalid:false|x:b2,q|y:bin(balance,q,sort={"field":"balance","op":"mean","order":"descending"})'
+      );
+
+      assert.deepEqual(specQ, {
+        transform: {
+          calculate: [{field: 'b2', expr: '3*datum["b2"]'}],
+          filter: 'datum["b2"] > 60',
+          filterInvalid: false
+        },
+        mark: Mark.POINT,
+        encodings: [
+          {channel: Channel.X, field: 'b2', type: Type.QUANTITATIVE},
+          {bin: {}, channel: Channel.Y, field: 'balance', type: Type.QUANTITATIVE, sort: {field: 'balance', op: AggregateOp.MEAN, order: SortOrder.DESCENDING}}
+        ]
+      });
+    });
+  });
+
+  describe('shorthandParser', () => {
+    describe('encoding', () => {
+      it('should correctly parse an encoding query given a channel and fieldDefShorthand', () => {
+        let specQ: SpecQuery = {mark: Mark.POINT, encodings: []} as SpecQuery;
+
+        shorthandParser.encoding(
+          specQ,
+          'x',
+          'bin(a,q,maxbins=20,scale={"type":"log"},sort={"field":"a","op":"mean","order":"descending"})'
+        );
+
+        assert.deepEqual(specQ.encodings[0], {
+          bin: {maxbins: 20},
+          channel: Channel.X,
+          field: 'a',
+          type: Type.QUANTITATIVE,
+          scale: {type: ScaleType.LOG},
+          sort: {field: 'a', op: AggregateOp.MEAN, order: SortOrder.DESCENDING}
+        });
+      });
+    });
+
+    describe('fn', () => {
+      it('should correctly parse an encoding query given a fieldDefShorthand with aggregation function', () => {
+        let encQ: EncodingQuery = {} as EncodingQuery;
+        shorthandParser.fn(encQ, 'sum(horsepower,q)');
+        assert.deepEqual(encQ, {aggregate: AggregateOp.SUM, field: 'horsepower', type: Type.QUANTITATIVE});
+      });
+
+      it('should correctly parse an encoding query given a fieldDefShorthand with count function', () => {
+        let encQ: EncodingQuery = {} as EncodingQuery;
+        shorthandParser.fn(encQ, 'count(model,n)');
+        assert.deepEqual(encQ,{aggregate: AggregateOp.COUNT, field: 'model', type: Type.NOMINAL});
+      });
+
+      it('should correctly parse an encoding query given a fieldDefShorthand with timeunit function', () => {
+        let encQ: EncodingQuery = {} as EncodingQuery;
+        shorthandParser.fn(encQ, 'hours(a,q)');
+        assert.deepEqual(encQ, {field: 'a', timeUnit: TimeUnit.HOURS, type: Type.QUANTITATIVE});
+      });
+
+      it('should correctly parse an encoding query given a fieldDefShorthand with bin function', () => {
+        let encQ: EncodingQuery = {} as EncodingQuery;
+        shorthandParser.fn(encQ, 'bin(a,q,maxbins=20,sort={"field":"a","op":"mean","order":"descending"})');
+        assert.deepEqual(encQ, {
+          bin: {maxbins: 20},
+          field: 'a',
+          type: Type.QUANTITATIVE,
+          sort: {field: 'a', op: AggregateOp.MEAN, order: SortOrder.DESCENDING}
```

Comment:
You can't really sort bin with SortField.  You should create a separate test that sort either N or O field with SortField.